### PR TITLE
[CSL-823] Rework security workers

### PR DESCRIPTION
**Pull request by gromakovsky**

### CODE REVIEW

File: godtossing/Pos/Ssc/GodTossing/Workers.hs [hunk 1]
```diff
@@ -419,3 +431,46 @@ waitUntilSend msgTag epoch slMultiplier = do
...
+checkForIgnoredCommitmentsWorker
+    :: forall ctx m.
+       SscMode SscGodTossing ctx m
+    => (WorkerSpec m, OutSpecs)
+checkForIgnoredCommitmentsWorker = localWorker $ do
+    counter <- newTVarIO 0
+    void $ onNewSlot True (checkForIgnoredCommitmentsWorkerImpl counter)
+
+-- This worker checks whether our commitments appear in blocks. This
+-- check is done only if we actually should participate in
```
Comment (neongreen) [inline godtossing/Pos/Ssc/GodTossing/Workers.hs]: The comment should say what happens if this check fails (i.e. just emits a warning).

File: godtossing/Pos/Ssc/GodTossing/Workers.hs [hunk 2]
```diff
@@ -419,3 +431,46 @@ waitUntilSend msgTag epoch slMultiplier = do
...
+-- GodTossing. It's triggered if there are
+-- 'mdNoCommitmentsEpochThreshold' consequent epochs during which we
+-- had to participate in GodTossing, but our commitment didn't appear
+-- in blocks.
+checkForIgnoredCommitmentsWorkerImpl
+    :: forall ctx m. (SscMode SscGodTossing ctx m)
+    => TVar Word -> SlotId -> m ()
```
Comment (neongreen) [inline godtossing/Pos/Ssc/GodTossing/Workers.hs]: And here it'd be nice to have a haddock saying what this counter counts.

File: godtossing/Pos/Ssc/GodTossing/Workers.hs [hunk 3]
```diff
@@ -419,3 +431,46 @@ waitUntilSend msgTag epoch slMultiplier = do
...
+checkForIgnoredCommitmentsWorkerImpl counter SlotId {..}
+    -- It's enough to do this check once per epoch near the end of the epoch.
+    | getSlotIndex siSlot /= 9 * fromIntegral blkSecurityParam = pass
```
Comment (neongreen) [inline godtossing/Pos/Ssc/GodTossing/Workers.hs]: I think that `when (getSlotIndex siSlot == 9 * fromIntegral blkSecurityParam) $ ...` is a bit nicer, but whatever.

File: godtossing/Pos/Ssc/GodTossing/Workers.hs [hunk 4]
```diff
@@ -419,3 +431,46 @@ waitUntilSend msgTag epoch slMultiplier = do
...
+            ourId <- getOurStakeholderId
+            globalCommitments <-
+                getCommitmentsMap . view gsCommitments <$> gtGetGlobalState
+            unless (ourId `HM.member` globalCommitments) $ do
+                -- `modifyTVar'` returns (), hence not used
+                newCounterValue <-
+                    atomically $ do
+                        !x <- succ <$> readTVar counter
+                        x <$ writeTVar counter x
+                when (newCounterValue > mdNoCommitmentsEpochThreshold) $ do
```
Comment (neongreen) [inline godtossing/Pos/Ssc/GodTossing/Workers.hs]: Shouldn't you reset this counter at some point?

File: godtossing/Pos/Ssc/GodTossing/Workers.hs (hunk 2 above)
Comment (gromakovsky) [reply godtossing/Pos/Ssc/GodTossing/Workers.hs]: It's a private function of this module, so I think it's not necessary, regular comment should be enough (will add).

File: godtossing/Pos/Ssc/GodTossing/Workers.hs (hunk 1 above)
Comment (gromakovsky) [reply godtossing/Pos/Ssc/GodTossing/Workers.hs]: Ok.

File: godtossing/Pos/Ssc/GodTossing/Workers.hs (hunk 4 above)
Comment (gromakovsky) [reply godtossing/Pos/Ssc/GodTossing/Workers.hs]: I should.

**Comment by gromakovsky** [comment]
@neongreen it's ready for re-review.

**Pull request merged**